examples/secret-init-as-a-standalone-tool.md

```diff
@@ -0,0 +1,87 @@
+## Secret-init as a standalone tool
+**Multi-provider setup**
+
+## Prerequisites
+
+- Golang `>= 1.21`
+- Makefile
+- Docker compose
+
+## Environment setup
+
+```bash
+# Deploy a Vault instance
+make up
+```
+
+```bash
+# FILE PROVIDER SETUP
+
+# Create a folder for the example assets
+mkdir -p example
+
+# Create secrets for the file provider
+printf "secret-value" >> "example/secret-file"
+printf "super-secret-value" >> "example/super-secret-value"
+
+#NOTE: Optionally you can set a mount path for the file provider by using the FILE_MOUNT_PATH environment variable.
+```
+
+```bash
+# VAULT PROVIDER SETUP
+
+# Create a tokenfile
+printf $VAULT_TOKEN > "example/token-file"
+export VAULT_TOKEN_FILE=$PWD/example/token-file
+
+#NOTE: Secret-init can authenticate to Vault by supplying role/path credentials. 
+
+# Create secrets for the vault provider
+vault kv put secret/test/mysql MYSQL_PASSWORD=3xtr3ms3cr3t
+vault kv put secret/test/aws AWS_ACCESS_KEY_ID=secretId AWS_SECRET_ACCESS_KEY=s3cr3t
+
+#NOTE: If you would like to use secret-init in daemon mode to renew secrets in the background.
+
+#Set daemon mode
+
+export SECRET_INIT_DAEMON="true"
+```
+
+```bash
+# Export environment variables
+export FILE_SECRET_1=file:$PWD/example/secret-file
+export FILE_SECRET_2=file:$PWD/example/super-secret-value
+export MYSQL_PASSWORD=vault:secret/data/test/mysql#MYSQL_PASSWORD
+export AWS_SECRET_ACCESS_KEY=vault:secret/data/test/aws#AWS_SECRET_ACCESS_KEY
+export AWS_ACCESS_KEY_ID=vault:secret/data/test/aws#AWS_ACCESS_KEY_ID
+```
```

Comment:
```suggestion
```bash
## Define secrets to inject

# Export environment variables
export FILE_SECRET_1=file:$PWD/example/secret-file
export FILE_SECRET_2=file:$PWD/example/super-secret-value
export MYSQL_PASSWORD=vault:secret/data/test/mysql#MYSQL_PASSWORD
export AWS_SECRET_ACCESS_KEY=vault:secret/data/test/aws#AWS_SECRET_ACCESS_KEY
export AWS_ACCESS_KEY_ID=vault:secret/data/test/aws#AWS_ACCESS_KEY_ID
```
```